src/pf_types.h

```diff
@@ -790,6 +790,58 @@ typedef struct pf_alarm_fixed
     */
 } pf_alarm_fixed_t;
 
+typedef struct pf_alarm_err
+{
+   pnet_pnio_status_t pnio_status;
+   /*
+    * pf_alarm_err_t may be followed by alarm_payload:
+    *    vendor_device_error_info = VendorId, DeviceId, data*
+    */
+} pf_alarm_err_t;
+
+typedef struct pf_diag_std
+{
+   uint32_t ext_ch_add_value;
+   uint32_t qual_ch_qualifier;
+   uint16_t ch_nbr;
+   uint16_t ch_properties; /* Appears, direction, channelgroup etc */
+   uint16_t ch_error_type;
+   uint16_t ext_ch_error_type;
+} pf_diag_std_t;
+
+/* Make pf_diag_usi_t equal in size to pf_diag_std_t */
+#define PF_DIAG_MANUF_DATA_LEN sizeof (pf_diag_std_t)
+typedef struct pf_diag_usi
+{
+   uint8_t manuf_data[PF_DIAG_MANUF_DATA_LEN];
+} pf_diag_usi_t;
+
+/* This is the end of list designator. */
+#define PF_DIAG_IX_NULL UINT16_MAX
+
+typedef struct pf_diag_item
+{
+   /* The selector is the usi member (0x0000..0x7fff => usi, else std) */
+   union fmt
+   {
+      pf_diag_std_t std;
+      pf_diag_usi_t usi;
+   } fmt;
+
+   bool in_use;
+   uint16_t usi;  /* pf_usi_values_t */
+   uint16_t next; /* Next in list (array index) */
+} pf_diag_item_t;
+
+#define PF_ALARM_MAX_PAYLOAD_DATA_LEN sizeof (pf_diag_item_t)
+
+typedef struct pf_alarm_payload
+{
+   uint16_t usi;
+   uint16_t len;
+   uint8_t data[PF_ALARM_MAX_PAYLOAD_DATA_LEN];
```

Comment:
in `pnet_alarm_send_process_alarm()` it says that the max payload length is 1408 bytes. Maybe some compile-time define could be used somehow?

Reply:
To better handle process alarms with large alarm payload, the stack maybe should not copy the data, but require that the application keeps the alarm data until the confirmation (callback) is received?